include/sisl/fds/stream_tracker.hpp

```diff
@@ -88,6 +88,18 @@ class StreamTracker {
         m_comp_slot_bits.set_bits(start_bit, end_idx - start_idx + 1);
     }
 
+    void rollback(int64_t new_end_idx) {
+        folly::SharedMutexWritePriority::ReadHolder holder(m_lock);
+        if ((new_end_idx < m_slot_ref_idx) ||
+            (new_end_idx >= (m_slot_ref_idx + int64_cast(m_active_slot_bits.size())))) {
+            throw std::out_of_range("Slot idx is not in range");
+        }
+
+        auto new_end_bit = new_end_idx - m_slot_ref_idx;
```

Comment:
This is not related to this PR, but just want to learn the reason behind `m_slot_ref_idx`:

by looking at the way we update (with `+=`, instead of `=`) `m_slot_ref_idx` in `truncate`, is there a reason we don't need to worry `m_slot_ref_idx` would exceeding the total size of bits?

Reply:
`m_slot_ref_idx` refers to the current starting active slot number. So anytime we truncate we move the slot_ref_idx. So if we have a stream with id=0, 1, 2, 3,....200 and so bit 0 represents id=0, bit 1 represents id=1 so on and so forth. Later say we truncate upto 80, then m_slot_ref_idx = 80 and then id=81 is represented by bit 0, bit 1 etc..

Reply:
okay, yes, after looking through how we calculate `upto_bit`, it makes sense (e.g. it should never exceed total bits). Though `upto_bit` looks like a pos, but it actually serves as a counter. Anyway it is a minor comment.